src/server/ApolloServer.ts

```diff
@@ -52,11 +58,32 @@ export class ApolloServer {
     this.server.use(bodyParser());
 
     const schema = await buildSchema({
-      resolvers: [ItemResolver],
+      globalMiddlewares: [DataLoaderMiddleware],
+      resolvers: [
+        path.resolve(__dirname, '../resolvers/**/resolver.ts'),
+      ],
+      authChecker,
       dateScalarMode: 'timestamp',
     });
 
-    const apolloServer = new ApolloServerKoa({ schema });
+    const apolloServer = new ApolloServerKoa({
+      schema,
+      tracing: true,
+      context: async ({ ctx }: { ctx: Koa.Context }) => {
+        const token = ctx.request.headers.authorization;
+        try {
+          const me = await this.jwt.verify<Context['me']>(token);
+          return {
+            ...ctx,
+            me,
+          };
+        } catch (e) {
+          return {
+            ...ctx,
```

Comment:
If I'm correct, it's okay to not return a copied ctx here.

Reply:
You are correct!!! 😂